### PR TITLE
rg-4: Add Common Inputs

### PR DESCRIPTION
**Pull request by Valentyn13**

### CODE REVIEW

**Comment by Valentyn13** [comment]
I have hardcoded width of the input and text-area in the styles, it will be necessary to remove this and replace it with some dynamic value. If adding width props to the components  is the right solution?

**Review by apofis96 — APPROVED** [review]

File: frontend/src/bundles/common/components/input/input.tsx [hunk 1]
```diff
@@ -23,17 +26,29 @@ const Input = <T extends FieldValues>({
     name,
     placeholder = '',
     type = 'text',
+    disabled = false,
 }: Properties<T>): JSX.Element => {
     const { field } = useFormController({ name, control });
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/input.tsx:31]: This component should not know anything about the outer scope

File: frontend/src/bundles/common/components/input/input.tsx [hunk 2]
```diff
@@ -23,17 +26,29 @@ const Input = <T extends FieldValues>({
...
     const hasError = Boolean(error);
-
     return (
-        <label>
-            <span>{label}</span>
-            <input {...field} type={type} placeholder={placeholder} />
-            {hasError && <span>{error as string}</span>}
+        <label className={styles.label}>
+            <span className={styles.labelName}>{label}</span>
+            <input
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/input.tsx:37]: I'd like to have BaseInput component, and FormGroup component with validation, label, etc...

File: frontend/src/bundles/common/components/input/input.tsx [hunk 3]
```diff
@@ -23,17 +26,29 @@ const Input = <T extends FieldValues>({
...
+                className={`
+                ${styles.input}
+                ${hasError && !disabled && styles.error}
+                ${!hasError && field.value.length > 0 && styles.filled}
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/input.tsx:41]: Avoid such conditions in layout

File: frontend/src/bundles/common/components/text-area/text-area.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    const hasError = Boolean(error);
+    return (
+        <label className={styles.label}>
+            <span>{label}</span>
+            <textarea
+                className={`
+            ${styles.textArea}
+            ${disabled && styles.disabled}
+            ${hasError && !disabled && styles.error}
+            ${!hasError && field.value.length > 0 && styles.filled}
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/text-area/text-area.tsx:39]: Looks like copy-paste

File: frontend/src/bundles/common/components/input/input.tsx [hunk 5]
```diff
@@ -23,17 +26,29 @@ const Input = <T extends FieldValues>({
...
+                ${disabled && styles.disabled}
+                `}
+                {...field}
+                type={type}
+                placeholder={placeholder}
+                disabled={disabled}
+            />
+            {hasError && !disabled && (
+                <span className={styles.errorText}>{error as string}</span>
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/input.tsx:50]: Avoid using `as` operator

File: frontend/src/bundles/common/components/input/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,64 @@
+.label {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    width: 280px;
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/styles.module.scss:5]: Why `width` is fixed?

File: frontend/src/bundles/common/components/input/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    color: #ccd7dc;
+}
+
+.filled {
+    color: #302e36;
+    border: 1px solid #768fbe;
+}
+
+.errorText {
+    position: absolute;
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/components/input/styles.module.scss:51]: Try to find better solution, without `position: absolute;`

**Review by Vitaliy-Kalenichenko — CHANGES_REQUESTED** [review]
TextArea component looks like copy-paste of Input component. This is considered bad practice

**Comment by Valentyn13** [comment]
usage of inputs

**Comment by Vitaliy-Kalenichenko** [comment]
Looks great
Resolve conflicts

File: .github/CODEOWNERS
Comment (Vitaliy-Kalenichenko) [inline .github/CODEOWNERS:1]: Why this file has been removed?

File: .gitignore [hunk 8]
```diff
@@ -3,7 +3,7 @@ node_modules
 
 # production
 build
-.idea
```
Comment (Vitaliy-Kalenichenko) [inline .gitignore:6]: It would be nice to have it here

**Pull request closed without merging**